diff --git a/src/wasm/wasm-objects.cc b/src/wasm/wasm-objects.cc
index 91ebf45a..07c9d795 100644
--- a/src/wasm/wasm-objects.cc
+++ b/src/wasm/wasm-objects.cc
@@ -356,12 +356,7 @@ void WasmTableObject::SetFunctionTableEntry(Isolate* isolate,
     SBXCHECK_BOUNDS(func_index, module->functions.size());
     auto* wasm_function = module->functions.data() + func_index;
     UpdateDispatchTable(isolate, table, entry_index, wasm_function,
-                        target_instance_data
-#if V8_ENABLE_DRUMBRAKE
-                        ,
-                        func_index
-#endif  // V8_ENABLE_DRUMBRAKE
-    );
+                        target_instance_data);
   } else if (WasmJSFunction::IsWasmJSFunction(*external)) {
     UpdateDispatchTable(isolate, table, entry_index,
                         Cast<WasmJSFunction>(external));
@@ -539,12 +534,7 @@ bool FunctionSigMatchesTable(wasm::CanonicalTypeIndex sig_id,
 void WasmTableObject::UpdateDispatchTable(
     Isolate* isolate, DirectHandle<WasmTableObject> table, int entry_index,
     const wasm::WasmFunction* func,
-    DirectHandle<WasmTrustedInstanceData> target_instance_data
-#if V8_ENABLE_DRUMBRAKE
-    ,
-    int target_func_index
-#endif  // V8_ENABLE_DRUMBRAKE
-) {
+    DirectHandle<WasmTrustedInstanceData> target_instance_data) {
   DirectHandle<TrustedObject> implicit_arg =
       func->imported
           // The function in the target instance was imported. Use its imports
@@ -593,17 +583,11 @@ void WasmTableObject::UpdateDispatchTable(
         wasm::GetProcessWideWasmCodePointerTable()->GetEntrypoint(
             call_target, signature_hash),
         sig_id, signature_hash,
-#if V8_ENABLE_DRUMBRAKE
-        target_func_index,
-#endif
         wasm::GetWasmImportWrapperCache()->FindWrapper(call_target),
         WasmDispatchTable::kExistingEntry);
   } else {
     dispatch_table->SetForNonWrapper(entry_index, *implicit_arg, call_target,
                                      sig_id,
-#if V8_ENABLE_DRUMBRAKE
-                                     target_func_index,
-#endif
                                      WasmDispatchTable::kExistingEntry);
   }
 
@@ -694,9 +678,6 @@ void WasmTableObject::UpdateDispatchTable(
              Builtins::EntryOf(Builtin::kWasmToJsWrapperInvalidSig, isolate));
   dispatch_table->SetForWrapper(entry_index, *import_data, call_target, sig_id,
                                 signature_hash,
-#if V8_ENABLE_DRUMBRAKE
-                                WasmDispatchTable::kInvalidFunctionIndex,
-#endif  // V8_ENABLE_DRUMBRAKE
                                 wasm_code, WasmDispatchTable::kExistingEntry);
 }
 
@@ -742,9 +723,6 @@ void WasmTableObject::UpdateDispatchTable(
   SBXCHECK(FunctionSigMatchesTable(sig_index, dispatch_table->table_type()));
   dispatch_table->SetForWrapper(entry_index, implicit_arg, call_target,
                                 sig_index, wasm_code->signature_hash(),
-#if V8_ENABLE_DRUMBRAKE
-                                WasmDispatchTable::kInvalidFunctionIndex,
-#endif  // V8_ENABLE_DRUMBRAKE
                                 wasm_code, WasmDispatchTable::kExistingEntry);
 }
 
@@ -1528,9 +1506,7 @@ FunctionTargetAndImplicitArg::FunctionTargetAndImplicitArg(
 #endif  // V8_ENABLE_DRUMBRAKE
   } else {
     // The function in the target instance was not imported.
-#if V8_ENABLE_DRUMBRAKE
-    target_func_index_ = target_func_index;
-#endif  // V8_ENABLE_DRUMBRAKE
+/* V8_ENABLE_DRUMBRAKE disabled */
   }
   call_target_ = target_instance_data->GetCallTarget(target_func_index);
 }
@@ -1559,9 +1535,6 @@ void ImportedFunctionEntry::SetGenericWasmToJs(
 
   instance_data_->dispatch_table_for_imports()->SetForWrapper(
       index_, *import_data, wrapper_entry, sig_id, sig->signature_hash(),
-#if V8_ENABLE_DRUMBRAKE
-      WasmDispatchTable::kInvalidFunctionIndex,
-#endif  // V8_ENABLE_DRUMBRAKE
       nullptr, WasmDispatchTable::kNewEntry);
 #if V8_ENABLE_DRUMBRAKE
   instance_data_->imported_function_indices()->set(index_, -1);
@@ -1595,9 +1568,6 @@ void ImportedFunctionEntry::SetCompiledWasmToJs(
       v8_flags.wasm_jitless ? Address{}
                             : wasm_to_js_wrapper->instruction_start(),
       sig_id, sig->signature_hash(),
-#if V8_ENABLE_DRUMBRAKE
-      WasmDispatchTable::kInvalidFunctionIndex,
-#endif  // V8_ENABLE_DRUMBRAKE
       wasm_to_js_wrapper, WasmDispatchTable::kNewEntry);
 
 #if V8_ENABLE_DRUMBRAKE
@@ -1607,12 +1577,7 @@ void ImportedFunctionEntry::SetCompiledWasmToJs(
 
 void ImportedFunctionEntry::SetWasmToWasm(
     Tagged<WasmTrustedInstanceData> target_instance_data,
-    WasmCodePointer call_target, wasm::CanonicalTypeIndex sig_id
-#if V8_ENABLE_DRUMBRAKE
-    ,
-    int exported_function_index
-#endif  // V8_ENABLE_DRUMBRAKE
-) {
+    WasmCodePointer call_target, wasm::CanonicalTypeIndex sig_id) {
   TRACE_IFT("Import Wasm 0x%" PRIxPTR "[%d] = {instance_data=0x%" PRIxPTR
             ", target=0x%" PRIxPTR "}\n",
             instance_data_->ptr(), index_, target_instance_data.ptr(),
@@ -1623,9 +1588,7 @@ void ImportedFunctionEntry::SetWasmToWasm(
       instance_data_->dispatch_table_for_imports();
   dispatch_table->SetForNonWrapper(index_, target_instance_data, call_target,
                                    sig_id,
-#if V8_ENABLE_DRUMBRAKE
-                                   WasmDispatchTable::kInvalidFunctionIndex,
-#endif  // V8_ENABLE_DRUMBRAKE
+/* V8_ENABLE_DRUMBRAKE disabled */
                                    WasmDispatchTable::kExistingEntry);
 
 #if V8_ENABLE_DRUMBRAKE
@@ -2456,9 +2419,6 @@ void WasmDispatchTableData::Remove(int index, WasmCodePointer call_target) {
 void WasmDispatchTable::SetForNonWrapper(int index, Tagged<Object> implicit_arg,
                                          WasmCodePointer call_target,
                                          wasm::CanonicalTypeIndex sig_id,
-#if V8_ENABLE_DRUMBRAKE
-                                         uint32_t function_index,
-#endif  // V8_ENABLE_DRUMBRAKE
                                          NewOrExistingEntry new_or_existing) {
   if (implicit_arg == Smi::zero()) {
     DCHECK_EQ(wasm::kInvalidWasmCodePointer, call_target);
@@ -2498,9 +2458,6 @@ void WasmDispatchTable::SetForWrapper(int index, Tagged<Object> implicit_arg,
                                       Address call_target,
                                       wasm::CanonicalTypeIndex sig_id,
                                       uint64_t signature_hash,
-#if V8_ENABLE_DRUMBRAKE
-                                      uint32_t function_index,
-#endif  // V8_ENABLE_DRUMBRAKE
                                       wasm::WasmCode* compiled_wrapper,
                                       NewOrExistingEntry new_or_existing) {
   DCHECK_NE(implicit_arg, Smi::zero());
@@ -2954,9 +2911,7 @@ bool WasmExportedFunction::IsWasmExportedFunction(Tagged<Object> object) {
   Tagged<JSFunction> js_function = Cast<JSFunction>(object);
   Tagged<Code> code = js_function->code(GetCurrentIsolateForSandbox());
   if (CodeKind::JS_TO_WASM_FUNCTION != code->kind() &&
-#if V8_ENABLE_DRUMBRAKE
-      code->builtin_id() != Builtin::kGenericJSToWasmInterpreterWrapper &&
-#endif  // V8_ENABLE_DRUMBRAKE
+/* V8_ENABLE_DRUMBRAKE disabled */
       code->builtin_id() != Builtin::kJSToWasmWrapper &&
       code->builtin_id() != Builtin::kWasmPromising &&
       code->builtin_id() != Builtin::kWasmStressSwitch) {
